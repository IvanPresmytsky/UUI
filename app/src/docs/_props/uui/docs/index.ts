export * from './common';
export * from './../components/pickers/common';
export * from './../components/iconPicker/IconPicker';
