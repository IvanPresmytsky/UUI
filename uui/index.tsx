import './assets/styles/typography.scss';

export * from './components';
export * from './i18n';
export * from './helpers';
export * from './settings';
export type * from './settings.types';
