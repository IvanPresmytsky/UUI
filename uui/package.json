{
  "name": "@epam/uui",
  "version": "5.1.2",
  "description": "EPAM UUI components set",
  "author": "EPAM",
  "license": "MIT",
  "main": "index.js",
  "epam:uui:main": "build/index.esm.js",
  "typings": "index.d.ts",
  "scripts": {
    "build": "node ../uui-build/scripts/buildModule",
    "prepublish": "yarn build"
  },
  "dependencies": {
    "@epam/assets": "5.1.2",
    "@epam/uui-components": "5.1.2",
    "@epam/uui-core": "5.1.2",
    "classnames": "2.2.6",
    "dayjs": "1.10.7",
    "lodash.sortby": "4.7.0",
    "react-focus-lock": "2.9.2",
    "react-popper": "2.3.0"
  },
  "peerDependencies": {
    "prop-types": "*",
    "react": ">=16.0.0",
    "react-dom": ">=16.0.0"
  },
  "repository": {
    "type": "git",
    "url": "https://github.com/epam/UUI"
  },
  "gitHead": "f0d58123e11e321e4417e3bba7dc6c5c9026998c",
  "publishConfig": {
    "directory": "build"
  }
}
