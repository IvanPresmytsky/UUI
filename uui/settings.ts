import type { Settings } from './settings.types';

export const settings: Settings = {
    sizes: {
        defaults: {
            badge: '36',
            button: '36',
            checkbox: '18',
            countIndicator: '24',
            linkButton: '36',
            numericInput: '36',
            radioInput: '18',
            switch: '18',
            tabButton: '48',
            tag: '36',
            text: '36',
            textArea: '36',
            textInput: '36',
            rangeDatePicker: '36',
            datePicker: '36',
            labeledInput: '36',
            statusIndicator: '24',
            paginator: '30',
        },
        tag: {
            countIndicator: {
                18: '12',
                24: '18',
                30: '18',
                36: '18',
                42: '24',
                48: '24',
            },
        },
        pickerInput: {
            toggler: {
                defaults: {
                    size: '36',
                    tag: '36',
                },
                tag: {
                    24: '18',
                    30: '18',
                    36: '24',
                    42: '30',
                    48: '36',
                    60: '42',
                },
            },
            body: {
                dropdown: {
                    height: 300,
                    width: 360,
                    padding: '12',
                    row: {
                        default: '36',
                        cell: {
                            text: {
                                24: '24',
                                30: '30',
                                36: '36',
                                42: '42',
                                48: '48',
                                60: '48',
                            },
                            icon: {
                                24: '12',
                                30: '18',
                                36: '18',
                                42: '24',
                                48: '24',
                                60: '24',
                            },
                            isBoldSelectionIcon: {
                                24: true,
                            },
                            item: { // max 48, setup by row with 60????
                                default: '36',
                                verticalPadding: { // TODO: try to move to size css classes, compare different components
                                    24: '3',
                                    30: '6',
                                    36: '9',
                                    42: '6',
                                    48: '9',
                                },
                                avatar: {
                                    rest: {
                                        24: '18',
                                        30: '24',
                                        36: '30',
                                        42: '36',
                                        48: '36', // no design
                                    },
                                    multiline: {
                                        24: '30',
                                        30: '30',
                                        36: '36',
                                        42: '42',
                                        48: '48', // no design
                                    },
                                },
                            },
                        },
                    },
                    footer: {
                        switch: {
                            24: '12',
                            30: '18',
                            36: '18',
                            42: '24',
                            48: '24',
                        },
                        linkButton: {
                            24: '24',
                            30: '30',
                            36: '36',
                            42: '42',
                            48: '48',
                        },
                    },
                },
                mobile: {
                    header: {
                        titleSize: '48',
                    },
                    footer: {
                        linkButton: '48',
                    },
                    row: '48',
                    searchInput: '48',
                },
                modal: {
                    row: '36',
                    padding: '24',
                },
            },
        },
        rowAddons: {
            defaults: {
                icon: '18',
                indentUnit: 24,
                indentWidth: 12,
            },
            checkbox: {
                24: '12',
                30: '18',
                36: '18',
                42: '18',
                48: '18',
                60: '18',
            },
            icon: {
                24: '12',
                30: '18',
                36: '18',
                42: '24',
                48: '24',
                60: '24',
            },
            indentUnit: {
                24: 6,
                30: 12,
                36: 12,
                42: 24,
                48: 24,
                60: 24,
            },
            indentWidth: {
                24: 12,
                30: 18,
                36: 18,
                42: 24,
                48: 24,
                60: 24,
            },
        },
        text: {
            18: { lineHeight: 12, fontSize: 10 },
            24: { lineHeight: 18, fontSize: 12 },
            30: { lineHeight: 18, fontSize: 14 },
            36: { lineHeight: 18, fontSize: 14 },
            42: { lineHeight: 24, fontSize: 16 },
            48: { lineHeight: 24, fontSize: 16 },
            60: { lineHeight: 30, fontSize: 24 },
        },
        filtersPanel: {
            pickerInput: {
                body: {
                    default: '36',
                },
            },
        },
        labeledInput: {
            fillIcon: ['24', '30'],
        },
        badge: {
            countIndicator: {
                18: '12',
                24: '18',
                30: '18',
                36: '18',
                42: '24',
                48: '24',
            },
        },
        dataTable: {
            header: {
                row: {
                    default: '36',
                    cell: {
                        defaults: {
                            size: '36',
                            resizeMarker: '12',
                            padding: '12',
                            paddingEdge: '24',
                        },
                        checkbox: {
                            36: '18',
                            48: '18',
                            60: '18',
                        },
                        columnCaption: {
                            fontSize: '14',
                            fontWeight: '400',
                            size: '30',
                            uppercase: '12',
                        },
                        iconSize: {
                            36: '18',
                            48: '18',
                            60: '18',
                        },
                        truncate: ['48'],
                    },
                },
            },
            body: {
                row: {
                    default: '36',
                    cell: {
                        defaults: {
                            size: '36',
                            padding: '12',
                            paddingEdge: '24',
                        },
                        text: {
                            18: '18',
                            24: '24',
                            30: '30',
                            36: '36',
                            42: '42',
                            48: '48',
                            60: '48',
                        },
                    },
                },
            },
            columnsConfigurationModal: {
                columnRow: '30',
                countIndicator: '18',
                subgroupIcon: '18',
                search: '30',
                width: '560',
            },
        },
        modal: {
            window: {
                defaults: {
                    width: '420',
                },
            },
        },
        tabButton: {
            countIndicator: {
                36: '18',
                48: '18',
                60: '18',
            },
        },
    },
};
