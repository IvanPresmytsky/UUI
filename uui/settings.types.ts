interface DefaultSizes {
    badge: string;
    button: string;
    checkbox: string;
    countIndicator: string;
    dataPickerCell: string;
    linkButton: string;
    numericInput: string;
    pickerToggler: string;
    pickerItem: string;
    radioInput: string;
    switch: string;
    tabButton: string;
    tag: string;
    text: string;
    textArea: string;
    textInput: string;
    rangeDatePicker: string;
    datePicker: string;
    labeledInput: string;
    statusIndicator: string;
    dataTableCell: string,
    dataTableRow: string,
    dataTableHeaderCell: string,
    dataTableHeaderRow: string,
    paginator: string,
}

interface Sizes {
    [size: string | number]: string | number;
}

interface TextSize {
    fontSize?: string | number;
    fontWeight?: string | number;
    lineHeight?: string | number;
    size?: string | number;
}

interface UppercaseTextSize {
    uppercase: string;
}

interface UtmostSize {
    utmost: string | number;
}

interface EditableSize {
    editable: string | number;
}

interface DefaultSize {
    default: string | number;
}

interface MobileSize {
    mobile: string | number;
}

interface ModalSize {
    modal: string | number;
}

interface TagSizes {
    countIndicator: Sizes;
}

interface MobileDropdownWrapperSizes {
    linkButton: string;
}

interface PickerInputRowSizes extends MobileSize, ModalSize {
    padding: ModalSize & DefaultSize;
}

interface PickerInputSizes {
    height: number;
    width: number;
    rowSize: PickerInputRowSizes;
}

interface PickerTogglerSizes {
    tag: Sizes;
}

interface DataPickerHeaderTextSizes {
    fontWeight: string;
    size: string;
}

interface DataPickerHeaderSizes {
    text: DataPickerHeaderTextSizes;
}

interface DataPickerBodySizes {
    flexCell: DefaultSize;
    searchInput: MobileSize,
}

interface PickerItemAvatarSizes {
    rest: Sizes;
    multiline: Sizes;
}

interface PickerItemSizes {
    avatar: PickerItemAvatarSizes;
    columnGap: string;
    verticalPadding: Sizes;
}

interface DataPickerCellIsBoldIcon {
    [size: string | number]: boolean;
}

interface DataPickerCellSizes {
    isBoldIcon: DataPickerCellIsBoldIcon;
    padding: DefaultSize;
    paddingLeft: DefaultSize;
    text: Sizes;
    icon: Sizes;
}

interface DataPickerRowSizes {
    padding: DefaultSize;
    dataPickerCell: DefaultSize;
}

interface DataPickerFooterSizes {
    flexRowPadding: string;
    switch: Sizes;
    linkButton: Sizes & MobileSize;
}

interface RowAddonsSizes {
    checkbox: Sizes;
    icon: Sizes & DefaultSize;
    indentUnit: Sizes & DefaultSize;
    indentWidth: Sizes & DefaultSize;
}

interface TextSizes {
    [size: string | number]: TextSize;
}

interface FilterPickerBodySizes {
    pickerItem: string,
    dataPickerRow: string,
    dataPickerFooter: string,
    searchSize: string,
}

interface LabeledInputSizes {
    fillIcon: string[];
}

interface BadgeSizes {
    countIndicator: Sizes;
}

interface DataTableHeaderCellTooltipContentSizes {
    caption: TextSize;
    info: TextSize;
}

interface DataTableHeaderCellSizes {
    checkbox: Sizes;
    columnCaption: TextSize & UppercaseTextSize;
    iconSize: Sizes;
    resizeMarker: DefaultSize;
    leftPadding: DefaultSize & UtmostSize;
    rightPadding: DefaultSize & UtmostSize;
    tooltip: DataTableHeaderCellTooltipContentSizes;
}

interface DataTableHeaderRowSizes {
    iconSize: Sizes;
    truncate?: string[];
}

interface DataTableCellSizes {
    text: Sizes;
    leftPadding: EditableSize & DefaultSize & UtmostSize;
    rightPadding: EditableSize & DefaultSize & UtmostSize;
}

interface DataTableRowSizes {
    columnsGap: DefaultSize;
}

interface DataTableColumnsConfigurationModal {
    columnRowSize: string;
    columnGap: string;
    padding: string;
    pinIconButtonGap: string;
    groupTitleSize: string;
    groupTitleFontSize: string;
    groupTitleFontWeight: string;
    subgroupTitleSize: string;
    subgroupTitleFontSize: string;
    subgroupTitleFontWeight: string;
    subgroupTitleIconSize: string;
    searchAreaSize: string;
    searchSize: string;
    noFoundTitleSize: string;
    noFoundTitleFontSize: string;
    noFoundTitleFontWeight: string;
    noFoundSubTitleSize: string;
    noFoundSubTitleFontSize: string;
    noFoundSubTitleFontWeight: string;
}

interface PaginatorSizes {}

interface SizesSettings {
    defaults: DefaultSizes;
    tag: TagSizes;
    mobileDropdownWrapper: MobileDropdownWrapperSizes;
    pickerInput: PickerInputSizes;
    pickerToggler: PickerTogglerSizes;
    dataPickerHeader: DataPickerHeaderSizes;
    dataPickerBody: DataPickerBodySizes;
    pickerItem: PickerItemSizes;
    dataPickerCell: DataPickerCellSizes;
    dataPickerRow: DataPickerRowSizes;
    dataPickerFooter: DataPickerFooterSizes;
    rowAddons: RowAddonsSizes;
    text: TextSizes;
    filterPickerBody: FilterPickerBodySizes;
    labeledInput: LabeledInputSizes;
    badge: BadgeSizes;
    dataTableHeaderCell: DataTableHeaderCellSizes;
    dataTableHeaderRow: DataTableHeaderRowSizes;
    dataTableCell: DataTableCellSizes;
    dataTableRow: DataTableRowSizes;
    dataTableColumnsConfigurationModal: DataTableColumnsConfigurationModal;
    paginator: PaginatorSizes;
}

export interface Settings {
    /**
     * setting sizes for complex and compound components to support 'Size theming'
     */
    sizes: SizesSettings;
}
