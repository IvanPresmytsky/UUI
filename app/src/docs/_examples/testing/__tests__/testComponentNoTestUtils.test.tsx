/* eslint no-restricted-imports : 0 */
import React from 'react';
import { render, act, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { useUuiServices, StubAdaptedRouter, UuiContext, UuiContexts } from '@epam/uui-core';
import { TextInput } from '@epam/uui';

/** Reusable utils - START */
function UuiContextDefaultWrapper({ children }) {
    const testUuiCtx = {} as UuiContexts;
    const router = new StubAdaptedRouter();
    const { services } = useUuiServices({ router });
    Object.assign(testUuiCtx, services);
    return (
        <UuiContext.Provider value={ services }>
            { children }
        </UuiContext.Provider>
    );
}
async function renderSnapshot(reactElement: any) {
    const result = renderer.create(React.createElement(UuiContextDefaultWrapper, { children: reactElement }));
    await act(() => new Promise((resolve) => {
        setTimeout(resolve, 1);
    }));
    return result.toJSON();
}
async function renderToJsDom(reactElement: any) {
    const result = render(reactElement, { wrapper: UuiContextDefaultWrapper });
    await act(() => new Promise((resolve) => {
        setTimeout(resolve, 1);
    }));
    return result;
}
/** Reusable utils - END */

/** Start: This is some component which we are going to test. It's just an example. */
export interface SomeComponentProps {
    value?: string;
    onValueChange?: (value: string) => void;
}
export function SomeComponent(props: SomeComponentProps) {
    return (
        <TextInput value={ props.value } onValueChange={ props.onValueChange } />
    );
}
/** End */

describe('SomeComponent', () => {
    it('should render with minimum props', async () => {
        const tree = await renderSnapshot(<SomeComponent />);
        expect(tree).toMatchSnapshot();
    });
    it('should render textbox to jsdom', async () => {
        await renderToJsDom(<SomeComponent />);
        expect(screen.getAllByRole('textbox')).toBeDefined();
    });
});
