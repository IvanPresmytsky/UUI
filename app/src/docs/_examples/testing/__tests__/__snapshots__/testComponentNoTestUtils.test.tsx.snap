// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SomeComponent should render with minimum props 1`] = `
<div
  className="container uui-input-box -clickable root size-36 mode-form"
  onBlur={[Function]}
  onFocus={[Function]}
  tabIndex={-1}
>
  <input
    className="uui-input"
    onChange={[Function]}
    onKeyDown={[Function]}
    tabIndex={0}
    type="text"
    value=""
  />
</div>
`;
